src/ast/ast_rewrite_star_projections.c

```diff
@@ -0,0 +1,332 @@
+	for(uint i = 0; i < path_len; i += 2) {
+		const cypher_astnode_t *ast_node = cypher_ast_pattern_path_get_element(path, i);
+		const cypher_astnode_t *ast_alias = cypher_ast_node_pattern_get_identifier(ast_node);
+		if(ast_alias == NULL) continue;  // Unaliased node, do nothing.
+		// Add node alias to projection array.
+		array_append(*identifiers, ast_alias);
+	}
+
+	// Every odd offset corresponds to an edge.
+	for(uint i = 1; i < path_len; i += 2) {
+		const cypher_astnode_t *ast_edge = cypher_ast_pattern_path_get_element(path, i);
+		const cypher_astnode_t *ast_alias = cypher_ast_rel_pattern_get_identifier(ast_edge);
+		if(ast_alias == NULL) continue;  // Unaliased edge, do nothing.
+		// Add edge alias to projection array.
+		array_append(*identifiers, ast_alias);
+	}
+}
+
+static void _collect_aliases_in_pattern
+(
+	const cypher_astnode_t *pattern,
+	const cypher_astnode_t ***aliases
+) {
+	uint path_count = cypher_ast_pattern_npaths(pattern);
+	for(uint i = 0; i < path_count; i ++) {
+		_collect_aliases_in_path(cypher_ast_pattern_get_path(pattern, i), aliases);
+	}
+}
+
+static void _collect_with_projections(
+	const cypher_astnode_t *with_clause,
+	const cypher_astnode_t ***identifiers
+) {
+	uint projection_count = cypher_ast_with_nprojections(with_clause);
+	for(uint i = 0; i < projection_count; i ++) {
+		const cypher_astnode_t *projection = cypher_ast_with_get_projection(with_clause, i);
+		const char *alias = NULL;
+		const cypher_astnode_t *identifier = cypher_ast_projection_get_alias(projection);
+		if(identifier == NULL) {
+			// The projection was not aliased, so the projection itself must be an identifier.
+			identifier = cypher_ast_projection_get_expression(projection);
+			ASSERT(cypher_astnode_type(identifier) == CYPHER_AST_IDENTIFIER);
+		}
+		array_append(*identifiers, identifier);
+	}
+}
+
+// Sort an array of strings and remove duplicate entries.
+static void _uniqueArray(const cypher_astnode_t **arr) {
+#define MODIFIES_ISLT(a,b) (strcmp(cypher_ast_identifier_get_name(*a), \
+            cypher_ast_identifier_get_name(*b)) < 0)
+	int count = array_len(arr);
+	if(count == 0) return;
+	QSORT(const cypher_astnode_t *, arr, count, MODIFIES_ISLT);
+	uint unique_idx = 0;
+	for(int i = 0; i < count - 1; i ++) {
+		if(strcmp(cypher_ast_identifier_get_name(arr[i]),
+				  cypher_ast_identifier_get_name(arr[i + 1]))) {
+			arr[unique_idx++] = arr[i];
+		}
+	}
+	arr[unique_idx++] = arr[count - 1];
+	array_trimm_len(arr, unique_idx);
+}
+
+static void _collect_call_projections(
+	const cypher_astnode_t *call_clause,
+	const cypher_astnode_t ***aliases
+) {
+	uint yield_count = cypher_ast_call_nprojections(call_clause);
+	if(yield_count == 0) {
+		// If the procedure call is missing its yield part, include procedure outputs.
+		const char *proc_name = cypher_ast_proc_name_get_value(cypher_ast_call_get_proc_name(call_clause));
+		ProcedureCtx *proc = Proc_Get(proc_name);
+		ASSERT(proc);
```

Comment:
Because AST validation had yet to be performed we can't assume `proc` exists.
`GRAPH.QUERY g "call onon() RETURN *"`